x-pack/plugins/security_solution/server/lib/detection_engine/signals/utils.ts

```diff
@@ -514,17 +514,33 @@ export const createErrorsFromShard = ({ errors }: { errors: ShardError[] }): str
   });
 };
 
+/**
+ * Given a SignalSearchResponse this will return a valid last date if it can find one, otherwise it
+ * will return undefined.
+ * @param result The result to try and parse out the timestamp.
+ */
+export const lastValidDate = (result: SignalSearchResponse): Date | undefined => {
+  if (result.hits.hits.length === 0) {
+    return undefined;
+  } else {
+    const lastTimestamp = result.hits.hits[result.hits.hits.length - 1]._source['@timestamp'];
+    const isValid = lastTimestamp != null && moment(lastTimestamp).isValid();
+    if (!isValid) {
+      return undefined;
+    } else {
+      return new Date(lastTimestamp);
+    }
+  }
```

Comment:
I think this is a dangling else.

Reply:
I can remove it, but I think it's ok to be there. I like if/else/else if even if they're nested to make sure all the code paths are accounted for. Kind of the opposite of code golf a bit on this one.